### PR TITLE
[Cleanup] Cleaning up Raid.cpp

### PR DESCRIPTION
**Pull request by Aeadoin**
I've started to cleanup raid.cpp, I also resolved a couple potential crashes where we could have went out of bounds when returning GetPlayerIndex() that @Akkadius pointed out under #2782

### CODE REVIEW

File: utils/mods/legacy_combat.lua [hunk 1]
```diff
@@ -267,7 +267,7 @@ function TryCriticalHit(e)
 		return e;
 	end
 
-	if (self:IsPet() and self:GetOwner().valid and self:GetOwner():IsBot()) then
+	if (self:IsPet() and self:GetOwner().valid and self:GetOwner():is_bot()) then
```
Comment (Kinglykrab) [inline utils/mods/legacy_combat.lua:270]: `IsBot()` is valid, `is_bot()` is not.

File: zone/bot.cpp [hunk 2]
```diff
@@ -8255,7 +8248,7 @@ bool Bot::GetNeedsHateRedux(Mob *tar) {
 	//		return false;
 	//	}
 	//}
-	//else if (tar->IsBot()) {
+	//else if (tar->is_bot()) {
```
Comment (Kinglykrab) [inline zone/bot.cpp:8251]: Why was this changed?

File: zone/bot_raid.cpp [hunk 3]
```diff
@@ -92,34 +89,29 @@ void Raid::HandleBotGroupDisband(uint32 owner, uint32 gid)
...
-			bot_iter &&
-			gid != RAID_GROUPLESS &&
-			IsRaidMember(bot_iter->GetName()) &&
-			IsGroupLeader(bot_iter->GetName())
-			) {
-			auto r_group_members = GetRaidGroupMembers(GetGroup(bot_iter->GetName()));
-			auto group_inst = new Group(bot_iter);
+		if (b && gid != RAID_GROUPLESS && IsRaidMember(b->GetName()) && IsGroupLeader(b->GetName())) {
+			auto r_group_members = GetRaidGroupMembers(GetGroup(b->GetName()));
+			auto group_inst = new Group(b);
```
Comment (Kinglykrab) [inline zone/bot_raid.cpp:97]: Single letter? `g`?

File: zone/entity.cpp [hunk 4]
```diff
@@ -2243,7 +2243,7 @@ Raid* EntityList::GetRaidByBot(const Bot* bot)
 	std::list<RaidMember> rm;
 	auto GetMembersWhoAreBots = [&rm](Raid* r) -> std::list<RaidMember> {
 		for (auto const& m : r->members) {
-			if (m.IsBot) {
+			if (m.is_bot) {
```
Comment (Kinglykrab) [inline zone/entity.cpp:2246]: `IsBot`.

File: zone/lua_entity.cpp [hunk 5]
```diff
@@ -140,7 +140,7 @@ luabind::scope lua_register_entity() {
 	.def("CastToObject", &Lua_Entity::CastToObject)
 	.def("GetID", &Lua_Entity::GetID)
 	.def("IsBeacon", &Lua_Entity::IsBeacon)
-	.def("IsBot", &Lua_Entity::IsBot)
+	.def("is_bot", &Lua_Entity::IsBot)
```
Comment (Kinglykrab) [inline zone/lua_entity.cpp:143]: Why was this changed? Seems like a Replace All issue.

File: zone/npc.cpp [hunk 6]
```diff
@@ -322,7 +322,7 @@ NPC::NPC(const NPCType *npc_type_data, Spawn2 *in_respawn, const glm::vec4 &posi
 	AISpellVar.idle_no_sp_recast_max           = static_cast<uint32>(RuleI(Spells, AI_IdleNoSpellMaxRecast));
 	AISpellVar.idle_beneficial_chance          = static_cast<uint8> (RuleI(Spells, AI_IdleBeneficialChance));
 
-	// It's possible for IsBot() to not be set yet during Bot loading, so have to use an alternative to catch Bots
+	// It's possible for is_bot() to not be set yet during Bot loading, so have to use an alternative to catch Bots
```
Comment (Kinglykrab) [inline zone/npc.cpp:325]: Why was this changed?

File: zone/perl_mob.cpp [hunk 7]
```diff
@@ -3221,7 +3221,7 @@ void perl_register_mob()
 	package.add("IsBeneficialAllowed", &Perl_Mob_IsBeneficialAllowed);
 	package.add("IsBerserk", &Perl_Mob_IsBerserk);
 	package.add("IsBlind", &Perl_Mob_IsBlind);
-	package.add("IsBot", &Perl_Mob_IsBot);
+	package.add("is_bot", &Perl_Mob_IsBot);
```
Comment (Kinglykrab) [inline zone/perl_mob.cpp:3224]: Why was this changed?

File: zone/raids.h [hunk 8]
```diff
@@ -88,8 +88,8 @@ struct RaidMember{
 	bool IsGroupLeader;
 	bool IsRaidLeader;
 	bool IsLooter;
-	bool IsBot = false;
-	bool IsRaidMainAssistOne = false;
+	bool is_bot = false;
+	bool is_raid_main_assist_one = false;
```
Comment (Kinglykrab) [inline zone/raids.h:92]: Why were only these 2 changed?

File: zone/raids.h [hunk 9]
```diff
@@ -115,12 +115,12 @@ class Raid : public GroupIDConsumer {
...
 
 	void	AddMember(Client *c, uint32 group = 0xFFFFFFFF, bool rleader=false, bool groupleader=false, bool looter=false);
-	void	AddBot(Bot* b, uint32 group = 0xFFFFFFFF, bool rleader=false, bool groupleader=false, bool looter=false); 
+	void	AddBot(Bot* b, uint32 group = 0xFFFFFFFF, bool raid_leader=false, bool group_leader=false, bool looter=false);
 	void	RaidGroupSay(const char* msg, const char* from, uint8 language, uint8 lang_skill);
 	void	RaidSay(const char* msg, const char* from, uint8 language, uint8 lang_skill);
 	bool	IsEngaged();
-	Mob*	GetRaidMainAssistOneByName(const char* name);
-	void	RemoveMember(const char *c);
+	Mob* GetRaidMainAssistOne();
```
Comment (Kinglykrab) [inline zone/raids.h:122]: Formatting.

File: zone/raids.cpp [hunk 10]
```diff
@@ -2089,14 +2094,15 @@ void Raid::RaidSay(const char* msg, const char* from, uint8 language, uint8 lang
...
-				if (strlen(members[x].membername) > 0 && strcmp(members[x].membername, members[i].membername) != 0) {
-					SetRaidLeader(members[i].membername, members[x].membername);
-					UpdateRaidAAs();
-					SendAllRaidLeadershipAA();
-					break;
-				}
+		for (const auto& m : members) {
+			if (m.is_bot) {
+				continue;
+			}
```
Comment (Kinglykrab) [inline zone/raids.cpp:2100]: New line here maybe?

File: zone/raids.cpp [hunk 11]
```diff
@@ -2021,43 +2021,47 @@ bool Raid::DoesAnyMemberHaveExpeditionLockout(
...
-			return 1;
+bool Raid::IsEngaged()
+{
+
+	for (const auto& m : GetMembers()) {
+		if (m.is_bot) {
+			continue;
+		}
+		if (m.member && (m.member->IsEngaged() || m.member->GetAggroCount() > 0)) {
+			return true;
```
Comment (Kinglykrab) [inline zone/raids.cpp:2043]: Newline between maybe?

File: zone/raids.cpp [hunk 12]
```diff
@@ -1948,34 +1949,34 @@ void Raid::QueueClients(Mob *sender, const EQApplicationPacket *app, bool ack_re
...
 				continue;
 			}
-			if (ignore_sender && members[i].member == sender) {
+			if (ignore_sender && m.member == sender) {
 				continue;
 			}
-			if (group_only && members[i].GroupNumber != group_id) {
+			if (group_only && m.GroupNumber != group_id) {
 				continue;
 			}
```
Comment (Kinglykrab) [inline zone/raids.cpp:1967]: Newlines here?

File: zone/raids.cpp [hunk 13]
```diff
@@ -1711,72 +1702,75 @@ void Raid::MemberZoned(Client *c)
...
-					outapp.SetOpcode(OP_MobEnduranceUpdate);
-					MobEnduranceUpdate_Struct *endurance_update = (MobEnduranceUpdate_Struct *)outapp.pBuffer;
-					endurance_update->endurance = mob->GetEndurancePercent();
-					members[x].member->QueuePacket(&outapp, false);
-				}
+	for (const auto& m : members) {
+		if (m.is_bot) {
+			continue;
+		}
+		if (m.member && (!mob->IsClient() || ((m.member != mob->CastToClient()) && (m.GroupNumber == group_id)))) {
```
Comment (Kinglykrab) [inline zone/raids.cpp:1761]: Newline between here?

File: zone/raids.cpp [hunk 14]
```diff
@@ -1530,7 +1519,10 @@ void Raid::SendGroupLeadershipAA(uint32 gid)
 void Raid::SendAllRaidLeadershipAA()
 {
 	for (const auto& m : members) {
-		if (m.member && !m.IsBot) {
+		if (m.is_bot) {
+			continue;
+		}
+		if (m.member) {
```
Comment (Kinglykrab) [inline zone/raids.cpp:1525]: Newline between here maybe?

File: zone/raids.cpp [hunk 15]
```diff
@@ -1481,9 +1467,9 @@ void Raid::SendRaidMOTD()
 	if (motd.empty())
 		return;
 
-	for (uint32 i = 0; i < MAX_RAID_MEMBERS; i++)
-		if (members[i].member)
-			SendRaidMOTD(members[i].member);
+	for (const auto& m : members)
+		if (m.member)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1471]: Brackets?

File: zone/raids.cpp [hunk 16]
```diff
@@ -1544,13 +1536,13 @@ void Raid::LockRaid(bool lockFlag)
     auto results = database.QueryDatabase(query);
 
 	locked = lockFlag;
-	if(lockFlag)
+	if (lockFlag)
 		SendRaidLock();
 	else
 		SendRaidUnlock();
```
Comment (Kinglykrab) [inline zone/raids.cpp:1542]: Brackets?

File: zone/raids.cpp [hunk 17]
```diff
@@ -1711,72 +1702,75 @@ void Raid::MemberZoned(Client *c)
...
+				client->QueuePacket(&outapp, false);
 			}
 		}
 	}
 }
 
 void Raid::SendHPManaEndPacketsFrom(Mob *mob)
 {
-	if(!mob)
+	if (!mob)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1744]: Brackets?

File: zone/raids.cpp [hunk 18]
```diff
@@ -1711,72 +1702,75 @@ void Raid::MemberZoned(Client *c)
 
 void Raid::SendHPManaEndPacketsTo(Client *client)
 {
-	if(!client)
+	if (!client)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1705]: Brackets?

File: zone/raids.cpp [hunk 19]
```diff
@@ -1686,7 +1677,7 @@ void Raid::VerifyRaid()
 
 void Raid::MemberZoned(Client *c)
 {
-	if(!c)
+	if (!c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1680]: Brackets?

File: zone/raids.cpp [hunk 20]
```diff
@@ -1521,7 +1507,10 @@ void Raid::SendGroupLeadershipAA(Client *c, uint32 gid)
 void Raid::SendGroupLeadershipAA(uint32 gid)
 {
 	for (const auto& m : members) {
-		if (m.member && m.GroupNumber == gid && !m.IsBot) {
+		if (m.is_bot) {
+			continue;
+		}
+		if (m.member && m.GroupNumber == gid) {
```
Comment (Kinglykrab) [inline zone/raids.cpp:1513]: Newlines between here?

File: zone/raids.cpp [hunk 21]
```diff
@@ -1427,11 +1413,11 @@ void Raid::SendRaidUnlockTo(Client *c)
 
 void Raid::SendGroupDisband(Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1416]: Brackets?

File: zone/raids.cpp [hunk 22]
```diff
@@ -1413,11 +1399,11 @@ void Raid::SendRaidLockTo(Client *c)
 
 void Raid::SendRaidUnlockTo(Client *c)
 {
-	if(!c)
+	if (!c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1402]: Brackets?

File: zone/raids.cpp [hunk 23]
```diff
@@ -1399,11 +1385,11 @@ void Raid::SendRaidUnlock()
 
 void Raid::SendRaidLockTo(Client *c)
 {
-	if(!c)
+	if (!c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1388]: Brackets?

File: zone/raids.cpp [hunk 24]
```diff
@@ -1349,23 +1337,21 @@ void Raid::SendGroupUpdate(Client *to)
 
 void Raid::GroupUpdate(uint32 gid, bool initial)
 {
-	if(gid > 11) {//ungrouped member doesn't need grouping.
+	if (gid >= MAX_RAID_GROUPS) {//ungrouped member doesn't need grouping.
 		return;
 	}
-	for (int x = 0; x < MAX_RAID_MEMBERS; x++)
+	for (const auto& m : members)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1343]: Newline between here?

File: zone/raids.cpp [hunk 25]
```diff
@@ -1305,40 +1293,40 @@ void Raid::SendGroupUpdate(Client *to)
...
-					index++;
+				if (strcmp(to->GetName(), m.membername) != 0) {
+					strn0cpy(gu->membername[i], m.membername, 64);
+					i++;
 				}
 			}
 		}
 	}
-	if(strlen(gu->leadersname) < 1){
+	if (strlen(gu->leadersname) < 1) {
```
Comment (Kinglykrab) [inline zone/raids.cpp:1329]: Newline between here?

File: zone/raids.cpp [hunk 26]
```diff
@@ -1295,7 +1283,7 @@ void Raid::SendMakeGroupLeaderPacket(const char *who) //13
 
 void Raid::SendMakeGroupLeaderPacketTo(const char *who, Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1286]: Brackets?

File: zone/raids.cpp [hunk 27]
```diff
@@ -1224,38 +1212,38 @@ void Raid::SendRaidMoveAll(const char* who)
 
 void Raid::SendBulkRaid(Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1215]: Brackets?

File: zone/raids.cpp [hunk 28]
```diff
@@ -1211,11 +1199,11 @@ void Raid::SendRaidMoveAll(const char* who)
...
 	Client *c = entity_list.GetClientByName(who);
 	SendRaidRemoveAll(who);
-	if(c)
+	if (c)
 		SendRaidCreate(c);
-	if(c){
+	if (c) {
 		SendBulkRaid(c);
-	if(IsLocked()) { SendRaidLockTo(c); }
+	if (IsLocked()) { SendRaidLockTo(c); }
```
Comment (Kinglykrab) [inline zone/raids.cpp:1206]: Brackets and newlines here?

File: zone/raids.cpp [hunk 29]
```diff
@@ -1189,19 +1177,19 @@ void Raid::SendRaidDisbandAll()
 
 void Raid::SendRaidMove(const char* who, Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1180]: Brackets?

File: zone/raids.cpp [hunk 30]
```diff
@@ -1120,15 +1108,15 @@ void Raid::SendRaidAddAll(const char *who)
 
 void Raid::SendRaidRemove(const char *who, Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1111]: Brackets?

File: zone/raids.cpp [hunk 31]
```diff
@@ -1069,7 +1057,7 @@ void Raid::SendRaidCreate(Client *to){
 
 void Raid::SendRaidAdd(const char *who, Client *to)
 {
-	if(!to)
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1060]: Brackets?

File: zone/raids.cpp [hunk 32]
```diff
@@ -1026,40 +1019,35 @@ bool Raid::IsRaidMember(Client* c)
...
 }
 
 /*
  * Packet Functions Start
  */
-void Raid::SendRaidCreate(Client *to){
-	if(!to)
+void Raid::SendRaidCreate(Client *to)
+{
+	if (!to)
```
Comment (Kinglykrab) [inline zone/raids.cpp:1046]: Brackets?

File: zone/raids.cpp [hunk 33]
```diff
@@ -783,18 +774,23 @@ void Raid::BalanceMana(int32 penalty, uint32 gid, float range, Mob* caster, int3
...
+	int manataken_tmp = 0;
+
+	for (const auto& m : members)
 	{
-		if (members[gi].member && !members[gi].IsBot) {
-			if (members[gi].GroupNumber == gid)
+		if (m.is_bot) {
+			continue;
+		}
+		if (m.member) {
```
Comment (Kinglykrab) [inline zone/raids.cpp:786]: Newline between here?

File: zone/raids.cpp [hunk 34]
```diff
@@ -664,11 +669,11 @@ uint32 Raid::GetTotalRaidDamage(Mob* other)
...
 
-	for (uint32 i = 0; i < MAX_RAID_MEMBERS; i++) {
-		if(!members[i].member)
+	for (const auto& m : members) {
+		if (!m.member)
 			continue;
-		if (other->CheckAggro(members[i].member))
-			total += other->GetHateAmount(members[i].member,true);
+		if (other->CheckAggro(m.member))
+			total += other->GetHateAmount(m.member,true);
```
Comment (Kinglykrab) [inline zone/raids.cpp:676]: Brackets and newline here?

File: zone/raids.cpp [hunk 35]
```diff
@@ -624,35 +629,35 @@ void Raid::CastGroupSpell(Mob* caster, uint16 spellid, uint32 gid)
 {
 	float range, distance;
 
-	if(!caster)
+	if (!caster)
```
Comment (Kinglykrab) [inline zone/raids.cpp:632]: Brackets?

File: zone/raids.cpp [hunk 36]
```diff
@@ -572,18 +575,18 @@ void Raid::RaidSay(const char *msg, Client *c, uint8 language, uint8 lang_skill)
 
 void Raid::RaidGroupSay(const char *msg, Client *c, uint8 language, uint8 lang_skill)
 {
-	if(!c)
+	if (!c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:578]: Brackets?

File: zone/raids.cpp [hunk 37]
```diff
@@ -572,18 +575,18 @@ void Raid::RaidSay(const char *msg, Client *c, uint8 language, uint8 lang_skill)
...
 		return;
 
-	uint32 groupToUse = GetGroup(c->GetName());
+	uint32 group_id_to_use = GetGroup(c->GetName());
 
-	if(groupToUse > 11)
+	if (group_id_to_use > 11)
```
Comment (Kinglykrab) [inline zone/raids.cpp:583]: Brackets?

File: zone/raids.cpp [hunk 38]
```diff
@@ -523,37 +526,37 @@ uint8 Raid::GroupCount(uint32 gid)
...
 }
 
 uint32 Raid::GetGroup(Client *c)
 {
-	for(int x = 0; x < MAX_RAID_MEMBERS; x++)
+	for (const auto& m : members)
 	{
-		if(members[x].member == c)
-			return members[x].GroupNumber;
+		if (m.member == c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:551]: Brackets?

File: zone/raids.cpp [hunk 39]
```diff
@@ -523,37 +526,37 @@ uint8 Raid::GroupCount(uint32 gid)
...
+			return m.GroupNumber;
 	}
 	return RAID_GROUPLESS;
 }
 
 void Raid::RaidSay(const char *msg, Client *c, uint8 language, uint8 lang_skill)
 {
-	if(!c)
+	if (!c)
```
Comment (Kinglykrab) [inline zone/raids.cpp:559]: Brackets?

File: zone/raids.cpp [hunk 40]
```diff
@@ -365,7 +367,7 @@ void Raid::SetRaidLeader(const char *wasLead, const char *name)
 	strn0cpy(leadername, name, 64);
 
 	Client *c = entity_list.GetClientByName(name);
-	if(c)
+	if (c)
 		SetLeader(c);
 	else
 		SetLeader(nullptr); //sanity check, should never get hit but we want to prefer to NOT crash if we do VerifyRaid and leader never gets set there (raid without a leader?)
```
Comment (Kinglykrab) [inline zone/raids.cpp:373]: Brackets?

File: zone/raids.cpp [hunk 41]
```diff
@@ -76,25 +76,26 @@ Raid::~Raid()
...
 		{
-			if(strlen(members[x].membername) == 0)
+			if (strlen(m.membername) == 0)
 				continue;
 			else
 				count++;
 		}
-		if(count == 0)
+		if (count == 0)
 			return false;
```
Comment (Kinglykrab) [inline zone/raids.cpp:92]: Brackets and newlines here?

File: utils/mods/legacy_combat.lua (hunk 1 above)
Comment (Aeadoin) [reply utils/mods/legacy_combat.lua:270]: Yeah, this got caught up in a renaming unfortunately of the member attribute. I'll fix these up.

File: zone/bot.cpp (hunk 2 above)
Comment (Aeadoin) [reply zone/bot.cpp:8251]: same as above, renamed due to no context since it's a comment, I ended up just removing the commend cause it's pointless

File: zone/raids.h (hunk 8 above)
Comment (Aeadoin) [reply zone/raids.h:92]: I'll do the others, was just going through the items Akk had pointed out at first.

**Review by Kinglykrab — APPROVED** [review]
